ee/server/service/service_teams.go

```diff
@@ -211,6 +211,29 @@ func (svc *Service) ListTeams(ctx context.Context, opt fleet.ListOptions) ([]*fl
 	return svc.ds.ListTeams(ctx, filter, opt)
 }
 
+func (svc *Service) ListAvailableTeamsForUser(ctx context.Context, user *fleet.User) ([]*fleet.Team, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Team{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	availableTeams := []*fleet.Team{}
+	if user.GlobalRole != nil {
+		allTeams, err := svc.ds.ListTeams(ctx, fleet.TeamFilter{User: user, IncludeObserver: true}, fleet.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		// TODO: Should the full Team struct be included here for global users or should it be cast to a smaller struct?
```

Comment:
Good question.

@chiiph `ListTeams` seems to count the hosts and users (`hosts` and `host_users` table) of the teams and then fetching its enroll secrets.

What does the UI need, maybe they need `Name` and `ID` only?

Reply:
@chiiph, just following up to see if you had any thoughts on the question from Lucas above. Currently the full struct is being returned for the `ListTeams` so keeping the full struct here potentially allows the UI to skip some of those separate API calls (for global users at least). Either way, perhaps it is something to consider in a separate ticket so we can move ahead and merge this one?

Reply:
Hm, I wouldn't add counts to /login and /me. It would make the API slower.

Reply:
I would rename the current `ListTeams` to `ListTeamsWithCounts` or `ListTeamsWithExtra`, and have a new `ListTeams` that only loads from `teams`.

Reply:
That sounds good to me.